src/UI/Art/SpriteArtist/SpriteInfoView.java

```diff
@@ -0,0 +1,125 @@
+import CommandingOfficers.Commander;
+import Engine.Driver;
+import Engine.Utils;
+import Engine.GameEvents.GameEventQueue;
+import UI.InfoController;
+import UI.MapView;
+
+/**
+ * InfoView paints the CO's full portrait and a box with info in it.
+ * The contents of the info are defined by inputs; current game status or static CO info are both used
+ */
+public class SpriteInfoView extends MapView // Extend MapView for getDrawableMap(). We don't actually draw it, but we need fog info.
+{
+  private InfoController myControl;
+  
+  /** Width of the visible space in pixels. */
+  private int viewWidth;
+  /** Height of the visible space in pixels. */
+  private int viewHeight;
+
+  public SpriteInfoView(InfoController control)
+  {
+    Dimension dims = Driver.getInstance().gameGraphics.getScreenDimensions();
+    viewWidth = dims.width;
+    viewHeight = dims.height;
+    myControl = control;
+  }
+
+  @Override
+  public Dimension getPreferredDimensions()
+  {
+    return new Dimension(viewWidth, viewHeight);
+  }
+
+  @Override
+  public void setPreferredDimensions(int width, int height)
+  {
+    // The user wants to use a specific amount of screen. Figure out how many tiles to draw for them.
+    viewWidth = width;
+    viewHeight = height;
+    // Let SpriteOptions know we are changing things.
+    SpriteOptions.setScreenDimensions(viewWidth, viewHeight);
+  }
+
+  @Override
+  public void render(Graphics g)
+  {
+    DiagonalBlindsBG.draw(g);
+
+    // Get the CO info menu.
+    int drawScale = SpriteOptions.getDrawScale();
+    
+    int paneOuterBuffer = 4*drawScale; // sets both outer border and frame border size
+    int paneHSize = (int) (viewWidth*0.7) - paneOuterBuffer*4; // width of the BG color area
+    int paneVSize = (int) (viewHeight   ) - paneOuterBuffer*4; // height ""
+
+    // Get the current menu selections.
+    Commander co = myControl.getSelectedCO();
+    InfoPage page = myControl.getSelectedPage();
+
+    // Draw the commander art.
+    BufferedImage COPic = SpriteLibrary.getCommanderSprites(myControl.getSelectedCOInfo().name).body;
+    // justify bottom/right
+    g.drawImage(COPic, viewWidth - COPic.getWidth()*drawScale, viewHeight - COPic.getHeight()*drawScale,
+        COPic.getWidth()*drawScale, COPic.getHeight()*drawScale, null);
+
+    if (null != co)
+      CommanderOverlayArtist.drawCommanderOverlay(g, co, false);
+    
+    // add the actual info
+    g.setColor(SpriteUIUtils.MENUFRAMECOLOR); // outer buffer
+    g.fillRect(  paneOuterBuffer,   paneOuterBuffer, paneHSize + 2*paneOuterBuffer, paneVSize + 2*paneOuterBuffer);
+    g.setColor(SpriteUIUtils.MENUBGCOLOR);    // inside of the pane
+    g.fillRect(2*paneOuterBuffer, 2*paneOuterBuffer, paneHSize                    , paneVSize                    );
+    
+    // TODO: consider drawing all pages as one big image, so the user can scroll smoothly through it regardless of screen size
+    switch (page.pageType)
+    {
+      case CO_HEADERS:
```

Comment:
(copied from the old PR)
I'd like to add income and unit count here, but I don't want to duplicate code and I'm not sure how best to split up the parsing that happens down below. Any ideas?

It's probably also worth caching the results of all these calculations so they aren't done every frame, but I'm not super worried about that at this point.

Reply:
So then you'd have income and unit count on more than one page?

To answer your question, my first thought is:
You could have `Utils.getGameStatusData` return an object instead of a formatted string. The object could even have methods to extract strings for each bit of info it knows about, or just call toString() to get all of them.
Then you can call `getGameStatusData` before the switch here and reference it wherever needed.